### DISCUSSION TITLE
Ios support Spatial Audio by airplay now

### DISCUSSION THREAD

**Discussion by profoundeye** [mikebrady/shairport-sync#1915] [Ideas]
Now, iOS 18's AirPlay already supports Spatial Audio (atmos). Will shairport-sync support this? This is a very interesting feature. Let me explain why.

Apple Music supports Spatial Audio and has atmos. To support multi-channel audio, an AVR is usually required for decoding. However, many old AVRs do not support AirPlay. They can only process multi-channel audio through HDMI input data. Considering their outdated code update capabilities, many of these devices will not be updated.

If shairport-sync can support obtaining multi-channel encoding via AirPlay, third-party devices can receive wireless data. Then, old AVR devices can obtain multi-channel support for AirPlay through USB or HDMI.

Currently, similar support can only be obtained through Apple TV.

**Comment by mikebrady** [comment]
Thanks for the interesting question. There was an announcement about Atmos and HomePods (?) but I haven't followed is up. What form does iOS 18's Atmos support take? As in, how do you get it to work?

**Comment by profoundeye** [comment]
I anticipate that it operates as follows:
Spatial Audio music works in Apple Music are transmitted to AirPlay terminal devices through AirPlay.
The terminal device transmits to a multi-channel  decoder via USB. 
After decoding by the decoder, it is transmitted to multi-channel speakers（for example ：5.1）.
It has been confirmed that this can work when the terminal device is an Apple TV.

**Reply by mikebrady** [reply]
Thanks for this.

Out of curiosity, I did a little experiment:
1. Start playing a track on an iPhone.
2. While playing, using AirPlay, transfer the output to a HomePod mini ("HPm").
3. While still playing, using AirPlay add a Shairport Sync ("SPS") device, so that the HPm and the SPS devices are now playing.

What's interesting about this is that the audio coming to the SPS device is _not_ coming from the iPhone: it's coming from the HPm.

So it looks as if, with iOS 18, Apple have changed things a bit. Now, if an iOS device directs output to an HPm, it effectively hands it off to the HPm; the HPm takes over and the iOS device is no longer involved. Presumably that's how Spatial Audio is handled -- the HPm would be the source, not the iOS device.
